src/main/kotlin/com/pubnub/api/crypto/cryptor/AesCbcCryptor.kt

```diff
@@ -55,12 +58,40 @@ class AesCbcCryptor(val cipherKey: String) : Cryptor {
         }
     }
 
-    override fun encryptStream(stream: InputStream): Result<EncryptedStreamData> {
-        TODO("Not yet implemented")
+    override fun encryptStream(stream: InputStream): EncryptedStreamData {
+        try {
+            val ivBytes: ByteArray = createRandomIv()
+            val ivSpec: AlgorithmParameterSpec = IvParameterSpec(ivBytes)
+            val keyBytes: ByteArray = getKeyAsShaInHexBytes(cipherKey)
+            val newKey = SecretKeySpec(keyBytes, "AES")
+            val cipher = Cipher.getInstance(CIPHER_TRANSFORMATION)
+            cipher.init(Cipher.ENCRYPT_MODE, newKey, ivSpec)
+            val cipheredStream = CipherInputStream(stream, cipher)
+            val cipheredBytes = cipheredStream.readBytes()
+
+            return EncryptedStreamData(
+                stream = SizedInputStream(
```

Comment:
I see that metadata from EncryptedStreamData are not use here, we don't store IV there  as in EncryptedData ?

Reply:
It's should be there I think